.circleci: Change conda image to be cuda specific

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51495 DO NOT MERGE, testing smaller conda images
* **#51494 .circleci: Change conda image to be cuda specific**

The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D26281958](https://our.internmc.facebook.com/intern/diff/D26281958)

seemethere committed:
.circleci: Change conda image to be cuda specific

The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

janeyx99 reviewed (approved):
Looks good to me

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D26281958](https://our.internmc.facebook.com/intern/diff/D26281958)

[ghstack-poisoned]

walterddr commented on .circleci/scripts/binary_linux_test.sh:
```diff
@@ -45,12 +49,18 @@ if [[ "$PACKAGE_TYPE" == conda ]]; then
-    if [[ "$DESIRED_CUDA" != 'cpu' ]]; then
+      retry conda install -c pytorch -y cpuonly
+    else
       # DESIRED_CUDA is in format cu90 or cu102
```
do we still have this format? (lol)

seemethere replied:
Yes unfortunately we do :/

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D26281958](https://our.internmc.facebook.com/intern/diff/D26281958)

[ghstack-poisoned]

seemethere committed:
Update on ".circleci: Change conda image to be cuda specific"


The overall `pytorch/conda-cuda` image was getting to a ridiculous size
of 36GB so this splits up that image into cuda specific ones to try and
reduce the amount of things we have to download.

coincides with: https://github.com/pytorch/builder/pull/634

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D26281958](https://our.internmc.facebook.com/intern/diff/D26281958)

[ghstack-poisoned]
